plot_continuous_monitor.py

```diff
@@ -128,93 +108,40 @@ def plot_continuous_monitor(filename, open_graphs=False):
-  for line in open(os.path.join(scripts_dir, "plot_disk_utilization_base.gp"), "r"):
-    new_line = line.replace("__OUT_FILENAME__", "%s_disk_utilization.pdf" % filename).replace(
-      "__NAME__", out_filename)
-    disk_plot_file.write(new_line)
-  disk_plot_file.close()
-
-  xvdf_plot_output_name = plot_single_disk(filename, out_filename, "xvdf", ["xvdb"], scripts_dir)
-  xvdb_plot_output_name = plot_single_disk(filename, out_filename, "xvdb", ["xvdf"], scripts_dir)
-
-  monotasks_plot_filename = "%s_monotasks.gp" % filename
-  monotasks_plot_file = open(monotasks_plot_filename, "w")
-  for line in open(os.path.join(scripts_dir, "plot_monotasks_base.gp"), "r"):
-    new_line = line.replace("__OUT_FILENAME__", "%s_monotasks.pdf" % filename).replace(
-      "__NAME__", out_filename)
-    monotasks_plot_file.write(new_line)
-  monotasks_plot_file.close()
-
-  memory_plot_filename = "%s_memory.gp" % filename
-  memory_plot_file = open(memory_plot_filename, "w")
-  for line in open(os.path.join(scripts_dir, "plot_memory_base.gp"), "r"):
-    new_line = line.replace("__OUT_FILENAME__", "%s_memory.pdf" % filename).replace(
-      "__NAME__", out_filename)
-    memory_plot_file.write(new_line)
-  memory_plot_file.close()
-
-  subprocess.check_call("gnuplot %s" % utilization_plot_filename, shell=True)
-  subprocess.check_call("gnuplot %s" % monotasks_plot_filename, shell=True)
-  subprocess.check_call("gnuplot %s" % disk_plot_filename, shell=True)
-  subprocess.check_call("gnuplot %s" % memory_plot_filename, shell=True)
+    data = {
+      'time': time - start,
+      'xvdf utilization': xvdf_total_utilization,
+      'xvdb utilization': xvdb_total_utilization,
+      'cpu utilization': cpu_total / 8.0, # Cores in machine?
+      'bytes received': bytes_received / 125000000.,
+      'bytes transmitted': bytes_transmitted / 125000000., # what is the 125000000 magic number?
+      'running compute monotasks': running_compute_monotasks,
+      'running monotasks': running_macrotasks,
+      'gc fraction': gc_fraction,
+      'outstanding network bytes': outstanding_network_bytes / (1024 * 1024),
+      'macrotasks in network': macrotasks_in_network,
+      'macrotasks in compute': macrotasks_in_compute,
+      'cpu system': cpu_system / 8.0,
+      'macrotasks in disk': macrotasks_in_disk,
+      'xvdf read throughput': xvdf_read_throughput,
+      'xvdf write throughput': xvdf_write_throughput,
+      'xvdb read throughput': xvdb_read_throughput,
+      'xvdb write throughput': xvdb_write_throughput,
+      'xvdf running disk monotasks': xvdf_running_disk_monotasks,
+      'xvdb running disk monotasks': xvdb_running_disk_monotasks,
+      'free heap memory': free_heap_memory / BYTES_PER_GIGABYTE,
+      'free off heap memory': free_off_heap_memory / BYTES_PER_GIGABYTE,
+      'local running macrotasks': local_running_macrotasks
+    }
+    continuous_monitor_data.append(data)
+    if use_gnuplot:
+      write_data(out_file, gnuplot_data_line(data))
```

Comment:
Can you pass in out_filename rather than constructing out_file in this function?